vic/drivers/cesm/src/cesm_put_data.c

```diff
@@ -91,16 +94,91 @@ vic_cesm_put_data()
-        // l2x_vic[i].l2x_Fall_fco2_lnd = 0;
-        // l2x_vic[i].l2x_Fall_flxdst1 = 0;
-        // l2x_vic[i].l2x_Fall_flxdst2 = 0;
-        // l2x_vic[i].l2x_Fall_flxdst3 = 0;
-        // l2x_vic[i].l2x_Fall_flxdst4 = 0;
-        // l2x_vic[i].l2x_Fall_flxvoc = 0;
+        l2x_vic[i].l2x_Fall_fco2_lnd = 0;
+	// for now, uncommenting these out 
+	// because this might have been the source 
+	// of the floating point error in RFR runs
+        l2x_vic[i].l2x_Fall_flxdst1 = 0;
+        l2x_vic[i].l2x_Fall_flxdst2 = 0;
+        l2x_vic[i].l2x_Fall_flxdst3 = 0;
+        l2x_vic[i].l2x_Fall_flxdst4 = 0;
+        l2x_vic[i].l2x_Fall_flxvoc = 0;
         l2x_vic[i].l2x_Flrl_rofliq = 0;
-        // l2x_vic[i].l2x_Flrl_rofice = 0;
+        l2x_vic[i].l2x_Flrl_rofice = 0;
+
+	// populate reference values 
+
+	// 10m wind, VIC: m/s, CESM: m/s
+        l2x_vic[i].l2x_Sl_u10 = out_data[i][OUT_WIND][0];
+
+	// 2m reference temperature, VIC: C, CESM: K
+        l2x_vic[i].l2x_Sl_tref = out_data[i][OUT_AIR_TEMP][0] + CONST_TKFRZ;
+
+	// 2m reference specific humidity, VIC: kg/kg, CESM: g/g
+        l2x_vic[i].l2x_Sl_qref = CONST_EPS * 
+					out_data[i][OUT_VP][0] / out_data[i][OUT_PRESSURE][0];
+
+	// band-specific quantities
+	// note that these include a veg correction (AreaFactor)
+	// that is already in the put_data routine
+
+	// temperature, VIC: K, CESM: K
+        l2x_vic[i].l2x_Sl_t = out_data[i][OUT_RAD_TEMP][0];
+
+        // albedo, VIC: fraction, CESM: fraction 
+	// Note: VIC does not partition its albedo, thus all types are
+	// the same value 
+        // force->NetShortAtmos net shortwave flux (+ down)
+        // SWup = force->shortwave[NR] - energy.NetShortAtmos
+        albedo = out_data[i][OUT_ALBEDO][0];
+                
+	// albedo: direct, visible
+	l2x_vic[i].l2x_Sl_avsdr = albedo;
+
+        // albedo: direct , near-ir
+        l2x_vic[i].l2x_Sl_anidr = albedo;
+
+        // albedo: diffuse, visible
+        l2x_vic[i].l2x_Sl_avsdf = albedo;
+
+        // albedo: diffuse, near-ir
+        l2x_vic[i].l2x_Sl_anidf = albedo;
+
+        // snow height, VIC: cm, CESM: m
+	// convert to VIC units
+        l2x_vic[i].l2x_Sl_snowh = out_data[i][OUT_SNOW_DEPTH][0] / CM_PER_M; 
+
+        // net shortwave, VIC: W/m2, CESM: W/m2
+        l2x_vic[i].l2x_Fall_swnet = out_data[i][OUT_SWNET][0];
+
+        // longwave up, VIC: W/m2, CESM: W/m2
+	// adjust sign for CESM sign convention
+        l2x_vic[i].l2x_Fall_lwup = -1 * 
+                                        (out_data[i][OUT_LWDOWN][0] -
+                                             out_data[i][OUT_LWNET][0]);
+
+	// turbulent heat fluxes
+        // latent heat, VIC: W/m2, CESM: W/m2
+        l2x_vic[i].l2x_Fall_lat = out_data[i][OUT_LATENT][0];
+
+        // sensible heat, VIC: W/m2, CESM: W/m2
+	// TO-DO: check sign in VIC 4 coupling, this is inconsistent with 
+	// the history file output sign 
+        l2x_vic[i].l2x_Fall_sen += -1 * out_data[i][OUT_SENSIBLE][0];
```

Comment:
Yes - it should be negative. In VIC4 we took care of the sign convention in the `lnd_comp_mct.F90` routine, but I think we want to do this in the VIC driver now.